$MirOS$
--- acconfig.h.orig	Sun Mar  4 22:32:43 2007
+++ acconfig.h	Sun Mar  4 23:35:32 2007
@@ -270,8 +270,15 @@
 /* Define if the X Window System is missing or not being used.  */
 #undef X_DISPLAY_MISSING
 
+/* Define if lex declares yytext as a char [] by default.  */
+#undef YYTEXT_CHAR
+
 /* Define if lex declares yytext as a char * by default, not a char[].  */
 #undef YYTEXT_POINTER
+
+/* Define if lex declares yytext as a unsigned char [] by default.  */
+#undef YYTEXT_UCHAR
+
 
 
 /* Leave that blank line there!!  Autoheader needs it.
--- acgeneral.m4.orig	Sun Mar  4 22:32:43 2007
+++ acgeneral.m4	Sun Mar  4 23:41:54 2007
@@ -889,11 +889,21 @@ AC_DEFUN(AC_CONFIG_AUX_DIRS,
 [ac_aux_dir=
 for ac_dir in ${GNUSYSTEM_AUX_DIR} $1; do
   if test -f $ac_dir/install-sh; then
-    ac_aux_dir=$ac_dir
+    case $ac_dir in
+changequote(, )dnl
+      /*|[A-z]:/*) ac_aux_dir=$ac_dir;;
+changequote([, ])dnl
+      *)  ac_aux_dir=`pwd`/$ac_dir;;
+    esac
     ac_install_sh="$ac_aux_dir/install-sh -c"
     break
   elif test -f $ac_dir/install.sh; then
-    ac_aux_dir=$ac_dir
+    case $ac_dir in
+changequote(, )dnl
+      /*|[A-z]:/*) ac_aux_dir=$ac_dir;;
+changequote([, ])dnl
+      *)  ac_aux_dir=`pwd`/$ac_dir;;
+    esac
     ac_install_sh="$ac_aux_dir/install.sh -c"
     break
   elif test -f $ac_dir/shtool; then
@@ -1339,10 +1349,13 @@ ifelse([$6], , , [  ac_prog_rejected=no
 dnl $ac_dummy forces splitting on constant user-supplied paths.
 dnl POSIX.2 word splitting is done only on the output of word expansions,
 dnl not every word.  This closes a longstanding sh security hole.
+dnl On MS-DOS, we should use test -x, but in general test -f is better
+dnl because some systems do not support test -x.
+  if test -z "$COMSPEC"; then ac_x=-f; else ac_x=-x; fi
   ac_dummy="ifelse([$5], , $PATH, [$5])"
   for ac_dir in $ac_dummy; do
     test -z "$ac_dir" && ac_dir=.
-    if test -f $ac_dir/$ac_word; then
+    if test $ac_x $ac_dir/$ac_word; then
 ifelse([$6], , , dnl
 [      if test "[$ac_dir/$ac_word]" = "$6"; then
         ac_prog_rejected=yes
@@ -1396,7 +1409,9 @@ set dummy $2; ac_word=[$]2
 AC_MSG_CHECKING([for $ac_word])
 AC_CACHE_VAL(ac_cv_path_$1,
 [case "[$]$1" in
-  /*)
+changequote(,)dnl
+  /*|[A-z]:/*)
+changequote([,])dnl
   ac_cv_path_$1="[$]$1" # Let the user override the test with a path.
   ;;
   ?:/*)
@@ -1407,10 +1422,13 @@ AC_CACHE_VAL(ac_cv_path_$1,
 dnl $ac_dummy forces splitting on constant user-supplied paths.
 dnl POSIX.2 word splitting is done only on the output of word expansions,
 dnl not every word.  This closes a longstanding sh security hole.
+dnl On MS-DOS, we should use test -x, but in general test -f is better
+dnl because some systems do not support test -x.
+  if test -z "$COMSPEC"; then ac_x=-f; else ac_x=-x; fi
   ac_dummy="ifelse([$4], , $PATH, [$4])"
   for ac_dir in $ac_dummy; do
     test -z "$ac_dir" && ac_dir=.
-    if test -f $ac_dir/$ac_word; then
+    if test $ac_x $ac_dir/$ac_word; then
       ac_cv_path_$1="$ac_dir/$ac_word"
       break
     fi
@@ -2050,7 +2068,7 @@ dnl ### Creating output files
 
 dnl AC_CONFIG_HEADER(HEADER-TO-CREATE ...)
 AC_DEFUN(AC_CONFIG_HEADER,
-[define(AC_LIST_HEADER, $1)])
+[define([AC_LIST_HEADER], $1)])
 
 dnl Link each of the existing files SOURCE... to the corresponding
 dnl link name in DEST...
@@ -2210,7 +2228,7 @@ EOF
 # platform that uses two characters for line-breaks (e.g., DOS), tr
 # would break.
 ac_LF_and_DOT="$(echo; echo .)"
-DEFS=$(sed -f conftest.defs confdefs.h | tr "$ac_LF_and_DOT" ' .')
+DEFS=$(sed -f conftest.defs confdefs.h | tr "\015$ac_LF_and_DOT" '  .')
 rm -f conftest.defs
 ])
 
@@ -2305,7 +2323,11 @@ changequote([, ])dnl
   .)  srcdir=.
       if test -z "$ac_dots"; then top_srcdir=.
       else top_srcdir=$(echo $ac_dots|sed 's%/$%%'); fi ;;
-  /*) srcdir="$ac_given_srcdir$ac_dir_suffix"; top_srcdir="$ac_given_srcdir" ;;
+changequote(,)dnl
+  /*|[A-z]:/*) # Absolute path.
+changequote([,])dnl
+    srcdir="$ac_given_srcdir$ac_dir_suffix"
+    top_srcdir="$ac_given_srcdir" ;;
   *) # Relative path.
     srcdir="$ac_dots$ac_given_srcdir$ac_dir_suffix"
     top_srcdir="$ac_dots$ac_given_srcdir" ;;
@@ -2313,8 +2335,9 @@ changequote([, ])dnl
 
 ifdef([AC_PROVIDE_AC_PROG_INSTALL],
 [  case "$ac_given_INSTALL" in
+  "") ;;
 changequote(, )dnl
-  [/$]*) INSTALL="$ac_given_INSTALL" ;;
+  [/$]*|[A-z]:/*) INSTALL="$ac_given_INSTALL" ;;
 changequote([, ])dnl
   *) INSTALL="$ac_dots$ac_given_INSTALL" ;;
   esac
@@ -2331,8 +2354,46 @@ changequote([, ])dnl
   *) ac_comsub= ;;
   esac
 
-  ac_file_inputs=$(echo $ac_file_in|sed -e "s%^%$ac_given_srcdir/%" -e "s%:% $ac_given_srcdir/%g")
-  sed -e "$ac_comsub
+  ac_file_inputs=`echo $ac_file_in|sed -e "s%:% $ac_given_srcdir/%g" -e "s%^%$ac_given_srcdir/%"`
+  # Replace lines of the form ac_include foo with the contents of foo:
+  # first, from the ac_include lines construct a list of file names.
+  # From that list, we construct a second list of those files that exist.
+  # Then we construct lists of sed commands for including those files,
+  # and a sed command that removes the ac_include lines. We don't attempt
+  # to read non-existent files because some (buggy) versions of sed choke
+  # on this. Have to use sed because old (Ultrix, SunOS) awk does not support
+  # getline or system. (Anyway, configure scripts aren't supposed to use awk.)
+  # Can't use only one -e and commands {dr foo} because foo has to be last.
+  # Be careful, because the filename may contain /.  Be careful with
+  # whitespace; we need to use echo.
+  # 
+  # No support for the multiple-output file :-separated stuff.
+  # 
+  file_substs=`sed -n \
+changequote(,)dnl
+'/^ac_include/s%ac_include[ 	]*\(.*\)%\1%p' \
+changequote([,])dnl
+               "$ac_given_srcdir/$ac_file_in"`
+  # Create the sed command line ...
+  file_subst_cmd='sed'
+  for ac_inc in $file_substs; do
+    if test -f "$ac_given_srcdir/$ac_inc"; then
+      ac_mung=`echo $ac_inc | sed 's,/,%,g'`
+      file_subst_cmd="$file_subst_cmd -e '/^ac_include $ac_mung$/r $ac_inc'"
+    fi
+  done
+  file_subst_cmd="$file_subst_cmd -e '/^ac_include/d'"
+  # ... and fix the whitespace and escaped slashes.
+  file_subst_cmd=`echo "$file_subst_cmd" | sed -e 's,%,\\\/,g' \
+changequote(,)dnl
+-e 's/ac_include /ac_include[ 	]*/g'`
+changequote([,])dnl
+  # cd into the srcdir because the files being included more or less
+  # must be part of the distribution. I can't find any way to do
+  # variable substitution in the sed commands (so the user could have,
+  # e.g., $top_srcdir in their ac_include line).
+  (cd $ac_given_srcdir && eval $file_subst_cmd $ac_file_in) \
+  | sed -e "$ac_comsub
 s%@configure_input@%$configure_input%g
 s%@srcdir@%$srcdir%g
 s%@top_srcdir@%$top_srcdir%g
@@ -2339,7 +2400,10 @@ s%@top_srcdir@%$top_srcdir%g
 ifdef([AC_PROVIDE_AC_PROG_INSTALL], [s%@INSTALL@%$INSTALL%g
 ])dnl
 dnl The parens around the eval prevent an "illegal io" in Ultrix sh.
-" $ac_file_inputs | (eval "$ac_sed_cmds") > $ac_file
+" | (eval "$ac_sed_cmds") > $ac_file
+dnl 
+dnl Had to remove $ac_file_inputs there to do the file substitutions.
+dnl 
 dnl This would break Makefile dependencies.
 dnl  if cmp -s $ac_file conftest.out 2>/dev/null; then
 dnl    echo "$ac_file is unchanged"
@@ -2363,7 +2427,7 @@ define(AC_OUTPUT_HEADER,
 #
 # ac_d sets the value in "#define NAME VALUE" lines.
 ac_dA='s%^\([ 	]*\)#\([ 	]*define[ 	][ 	]*\)'
-ac_dB='\([ 	][ 	]*\)[^ 	]*%\1#\2'
+ac_dB='\([ 	][ 	]*\)[^ 	].*$%\1#\2'
 ac_dC='\3'
 ac_dD='%g'
 # ac_u turns "#undef NAME" with trailing blanks into "#define NAME VALUE".
@@ -2399,7 +2463,7 @@ changequote([, ])dnl
   echo creating $ac_file
 
   rm -f conftest.frag conftest.in conftest.out
-  ac_file_inputs=$(echo $ac_file_in|sed -e "s%^%$ac_given_srcdir/%" -e "s%:% $ac_given_srcdir/%g")
+  ac_file_inputs=`echo $ac_file_in|sed -e "s%:% $ac_given_srcdir/%g" -e "s%^%$ac_given_srcdir/%"`
   cat $ac_file_inputs > conftest.in
 
 EOF
@@ -2567,7 +2631,7 @@ changequote([, ])dnl
 
   case "$srcdir" in
 changequote(, )dnl
-  [/$]*) ac_rel_source="$srcdir/$ac_source" ;;
+  [/$]*|[A-z]:/*) ac_rel_source="$srcdir/$ac_source" ;;
 changequote([, ])dnl
   *) ac_rel_source="$ac_dots$srcdir/$ac_source" ;;
   esac
@@ -2642,7 +2706,9 @@ changequote([, ])dnl
     case "$srcdir" in
     .) # No --srcdir option.  We are building in place.
       ac_sub_srcdir=$srcdir ;;
-    /*) # Absolute path.
+changequote(, )dnl
+    /*|[A-z]:/*) # Absolute path.
+changequote([, ])dnl
       ac_sub_srcdir=$srcdir/$ac_config_dir ;;
     *) # Relative path.
       ac_sub_srcdir=$ac_dots$srcdir/$ac_config_dir ;;
@@ -2649,7 +2715,9 @@ changequote([, ])dnl
     esac
 
     # Check for guested configure; otherwise get Cygnus style configure.
-    if test -f $ac_sub_srcdir/configure; then
+    if test -f ./configure; then
+      ac_sub_configure=./configure
+    elif test -f $ac_sub_srcdir/configure; then
       ac_sub_configure="$SHELL $ac_sub_srcdir/configure"
     elif test -f $ac_sub_srcdir/configure.in; then
       ac_sub_configure=$ac_configure
@@ -2663,14 +2731,17 @@ changequote([, ])dnl
 
       # Make the cache file name correct relative to the subdirectory.
       case "$cache_file" in
-      /*) ac_sub_cache_file=$cache_file ;;
+changequote(, )dnl
+      /*|[A-z]:/*) ac_sub_cache_file=$cache_file ;;
+changequote([, ])dnl
       *) # Relative path.
         ac_sub_cache_file="$ac_dots$cache_file" ;;
       esac
 ifdef([AC_PROVIDE_AC_PROG_INSTALL],
       [  case "$ac_given_INSTALL" in
+        "") ;;
 changequote(, )dnl
-        [/$]*) INSTALL="$ac_given_INSTALL" ;;
+        [/$]*|[A-z]:/*) INSTALL="$ac_given_INSTALL" ;;
 changequote([, ])dnl
         *) INSTALL="$ac_dots$ac_given_INSTALL" ;;
         esac
--- acspecific.m4.orig	Sun Mar  4 22:32:43 2007
+++ acspecific.m4	Sun Mar  4 23:37:01 2007
@@ -571,6 +571,7 @@ else
   AC_MSG_ERROR(cannot find output from $LEX; giving up)
 fi])
 LEX_OUTPUT_ROOT=$ac_cv_prog_lex_root
+rm -f "${LEX_OUTPUT_ROOT}.c"
 AC_SUBST(LEX_OUTPUT_ROOT)dnl
 
 AC_CACHE_CHECK(whether yytext is a pointer, ac_cv_prog_lex_yytext_pointer,
@@ -578,9 +579,19 @@ AC_CACHE_CHECK(whether yytext is a point
 # default is implementation-dependent. Figure out which it is, since
 # not all implementations provide the %pointer and %array declarations.
 ac_cv_prog_lex_yytext_pointer=no
-echo 'extern char *yytext;' >>$LEX_OUTPUT_ROOT.c
+# Avoid need for the lex library; very commonly, -lfl won't be found by
+# default, and thus the link below would fail if we didn't supply yywrap,
+# which would fool configure into thinking that yytext was not
+# a char * when it really is.
+echo '%%
+%%' | $LEX
+echo "extern char *yytext;
+#ifndef yywrap
+int yywrap () { return 1; }
+#endif" >>$LEX_OUTPUT_ROOT.c
 ac_save_LIBS="$LIBS"
 LIBS="$LIBS $LEXLIB"
+dnl Must use AC_TRY_LINK because yy.lex.c is not a function body.
 AC_TRY_LINK($(cat $LEX_OUTPUT_ROOT.c), , ac_cv_prog_lex_yytext_pointer=yes)
 LIBS="$ac_save_LIBS"
 rm -f "${LEX_OUTPUT_ROOT}.c"
@@ -588,7 +599,66 @@ rm -f "${LEX_OUTPUT_ROOT}.c"
 dnl
 if test $ac_cv_prog_lex_yytext_pointer = yes; then
   AC_DEFINE(YYTEXT_POINTER)
+else
+dnl
+AC_CACHE_CHECK(whether yytext is an array of char, ac_cv_prog_lex_yytext_char,
+[# POSIX says lex can declare yytext either as a pointer or an array; the
+# default is implementation-dependent. Figure out which it is, since
+# not all implementations provide the %pointer and %array declarations.
+ac_cv_prog_lex_yytext_char=no
+# Avoid need for the lex library; very commonly, -lfl won't be found by
+# default, and thus the link below would fail if we didn't supply yywrap,
+# which would fool configure into thinking that yytext was not
+# a char array when it really is.
+changequote(,)
+echo '%%
+%%' | $LEX
+echo "extern char yytext[];
+#ifndef yywrap
+int yywrap () { return 1; }
+#endif" >>$LEX_OUTPUT_ROOT.c
+changequote([,])
+ac_save_LIBS="$LIBS"
+LIBS="$LIBS $LEXLIB"
+dnl Must use AC_TRY_LINK because yy.lex.c is not a function body.
+AC_TRY_LINK(`cat $LEX_OUTPUT_ROOT.c`, , ac_cv_prog_lex_yytext_char=yes)
+LIBS="$ac_save_LIBS"
+rm -f "${LEX_OUTPUT_ROOT}.c"
+])
+dnl
+if test $ac_cv_prog_lex_yytext_char = yes; then
+AC_DEFINE(YYTEXT_CHAR)
+else
+dnl
+AC_CACHE_CHECK(whether yytext is an array of unsigned char, ac_cv_prog_lex_yytext_uchar,
+[# Some systems use an array of unsigned char for yytext.  Figure out if that
+# happens to be the case.
+ac_cv_prog_lex_yytext_uchar=no
+# Avoid need for the lex library; very commonly, -lfl won't be found by
+# default, and thus the link below would fail if we didn't supply yywrap,
+# which would fool configure into thinking that yytext was not
+# a unsigned char array when it really is.
+changequote(,)
+echo '%%
+%%' | $LEX
+echo "extern unsigned char yytext[];
+#ifndef yywrap
+int yywrap () { return 1; }
+#endif" >>$LEX_OUTPUT_ROOT.c
+changequote([,])
+ac_save_LIBS="$LIBS"
+LIBS="$LIBS $LEXLIB"
+dnl Must use AC_TRY_LINK because yy.lex.c is not a function body.
+AC_TRY_LINK(`cat $LEX_OUTPUT_ROOT.c`, , ac_cv_prog_lex_yytext_uchar=yes)
+LIBS="$ac_save_LIBS"
+rm -f "${LEX_OUTPUT_ROOT}.c"
+])
+dnl
+if test $ac_cv_prog_lex_yytext_uchar = yes; then
+AC_DEFINE(YYTEXT_UCHAR)
 fi
+fi
+fi
 ])
 
 AC_DEFUN(AC_PROG_INSTALL,
@@ -608,6 +678,8 @@ AC_MSG_CHECKING(for a BSD compatible ins
 if test -z "$INSTALL"; then
 AC_CACHE_VAL(ac_cv_path_install,
 [  IFS="${IFS= 	}"; ac_save_IFS="$IFS"; IFS=":"
+  # Hack for MSDOS and descendants.
+  if test -z "$COMSPEC"; then ac_x=-f; else ac_x=-x; fi
   for ac_dir in $PATH; do
     # Account for people who put trailing slashes in PATH elements.
     case "$ac_dir/" in
@@ -619,7 +691,7 @@ AC_CACHE_VAL(ac_cv_path_install,
       # Don't use installbsd from OSF since it installs stuff as root
       # by default.
       for ac_prog in ginstall scoinst install; do
-        if test -f $ac_dir/$ac_prog; then
+        if test $ac_x $ac_dir/$ac_prog; then
 	  if test $ac_prog = install &&
             grep dspmsg $ac_dir/$ac_prog >/dev/null 2>&1; then
 	    # AIX install.  It has an incompatible calling convention.
@@ -669,7 +741,9 @@ AC_DEFUN(AC_PROG_LN_S,
 [AC_MSG_CHECKING(whether ln -s works)
 AC_CACHE_VAL(ac_cv_prog_LN_S,
 [rm -f conftestdata
-if ln -s X conftestdata 2>/dev/null
+# MS-DOS is a special case, because it sort of pretends that ln -s
+# works for executables.
+if test -z "$COMSPEC" && ln -s X conftestdata 2>/dev/null
 then
   rm -f conftestdata
   ac_cv_prog_LN_S="ln -s"
@@ -1857,7 +1931,7 @@ AC_DEFUN(AC_FUNC_SELECT_ARGTYPES,
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
-extern select ($ac_cv_func_select_arg1,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg5);],,dnl
+extern int select ($ac_cv_func_select_arg1,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg5);],,dnl
         [ac_not_found=no ; break 3],ac_not_found=yes)
       done
      done
@@ -2631,12 +2705,14 @@ if test "$ac_x_libraries" = NO; then
 
   test -z "$x_direct_test_library" && x_direct_test_library=Xt
   test -z "$x_direct_test_function" && x_direct_test_function=XtMalloc
+  test -z "$x_direct_test_arguments" && x_direct_test_arguments=1
 
   # See if we find them without any special options.
   # Don't add to $LIBS permanently.
   ac_save_LIBS="$LIBS"
   LIBS="-l$x_direct_test_library $LIBS"
-AC_TRY_LINK(, [${x_direct_test_function}()],
+AC_TRY_LINK([#include <${x_direct_test_include}>],
+[${x_direct_test_function}(${x_direct_test_arguments})],
 [LIBS="$ac_save_LIBS"
 # We can link X programs with no special library path.
 ac_x_libraries=],
@@ -2840,6 +2916,32 @@ ac_cv_emxos2=yes, ac_cv_emxos2=no)])
 EMXOS2=
 test "$ac_cv_emxos2" = yes && EMXOS2=yes])
 
+dnl Check for djgpp.
+AC_DEFUN(AC_DJGPP,
+[AC_CACHE_CHECK(for djgpp environment, ac_cv_djgpp,
+[AC_TRY_COMPILE(,[
+#if defined (DJGPP) || defined (__DJGPP__)
+#define __HAVE_DJGPP__        1
+#endif
+return __HAVE_DJGPP__;],
+ac_cv_djgpp=yes, ac_cv_djgpp=no)
+rm -f conftest*])
+HAVE_DJGPP=
+test "$ac_cv_djgpp" = yes && HAVE_DJGPP=yes])
+
+dnl Check for win32.
+AC_DEFUN(AC_WIN32,
+[AC_CACHE_CHECK(for win32 environment, ac_cv_win32,
+[AC_TRY_COMPILE(,[
+#if defined(WIN32) || defined(__WIN32__) || defined(_WIN32)
+#define __HAVE_WIN32__        1
+#endif
+return __HAVE_WIN32__;],
+ac_cv_win32=yes, ac_cv_win32=no)
+rm -f conftest*])
+HAVE_WIN32=
+test "$ac_cv_win32" = yes && HAVE_WIN32=yes])
+
 dnl Check for the extension used for executables.  This knows that we
 dnl add .exe for Cygwin or mingw32.  Otherwise, it compiles a test
 dnl executable.  If this is called, the executable extensions will be
@@ -2874,6 +2976,25 @@ AC_MSG_RESULT(${ac_cv_exeext})
 dnl Setting ac_exeext will implicitly change the ac_link command.
 ac_exeext=$EXEEXT
 AC_SUBST(EXEEXT)])
+
+dnl Unset CC to run configure with cross compiler.
+AC_DEFUN(AC_UNSET_CC, [
+ZZ=
+if test "$cross_compiling" = yes &&
+   (test "x$CC" = "xdos-gcc" || test "x$CC" = "xi386-mingw32-gcc" || test "x$CC" = "xgnuwin32gcc") ; then
+ZZ=$CC
+unset CC
+cross_compiling=no
+fi
+])
+
+dnl Restore CC that has been unset by AC_UNSET_CC
+AC_DEFUN(AC_RESET_CC, [
+if test "x$ZZ" = "xdos-gcc" || test "x$ZZ" = "xi386-mingw32-gcc" || test "x$ZZ" = "xgnuwin32gcc" ; then
+CC=$ZZ
+cross_compiling=yes
+fi
+])
 
 
 dnl ### Checks for UNIX variants
