--- centericq-4.21.0.orig/debian/patches/jabber_segv.dpatch
+++ centericq-4.21.0/debian/patches/jabber_segv.dpatch
@@ -0,0 +1,38 @@
+#! /bin/sh -e
+## jabber_segv.dpatch by Julien LEMOINE <speedblue@debian.org>
+##
+## DP: applied patch from <temnota@kmv.ru> to fix segb in libjabber when registering new account
+
+[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
+patch_opts="${patch_opts:--f --no-backup-if-mismatch}"
+
+if [ $# -ne 1 ]; then
+    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+    exit 1
+fi
+case "$1" in
+       -patch) patch $patch_opts -p1 < $0;;
+       -unpatch) patch $patch_opts -p1 -R < $0;;
+        *)
+                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
+ent"
+                exit 1;;
+esac
+
+exit 0
+diff -bBdNrw -U5 centericq-4.21.0.orig/libjabber-0.1/jconn.c centericq-4.21.0/libjabber-0.1/jconn.c
+--- centericq-4.21.0.orig/libjabber-0.1/jconn.c	2005-08-26 13:07:00.000000000 +0200
++++ centericq-4.21.0/libjabber-0.1/jconn.c	2005-12-22 16:13:24.000000000 +0100
+@@ -344,11 +344,11 @@
+ {
+     fd_set fds;
+     struct timeval tv;
+     int r;
+ 
+-    if (!j || j->state == JCONN_STATE_OFF)
++    if (!j || j->state == JCONN_STATE_OFF || j->fd == -1)
+ 	return;
+ 
+     FD_ZERO(&fds);
+     FD_SET(j->fd, &fds);
+ 
