--- centericq-4.21.0.orig/debian/patches/remote_segv.dpatch
+++ centericq-4.21.0/debian/patches/remote_segv.dpatch
@@ -0,0 +1,32 @@
+#! /bin/sh -e
+## segv.dpatch by Julien LEMOINE <speedblue@debian.org>
+##
+## DP: apply patch from "Steve Langasek" <vorlon@debian.org> to fix remotely segfaultable DOS
+
+[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
+patch_opts="${patch_opts:--f --no-backup-if-mismatch}"
+
+if [ $# -ne 1 ]; then
+    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
+    exit 1
+fi
+case "$1" in
+       -patch) patch $patch_opts -p1 < $0;;
+       -unpatch) patch $patch_opts -p1 -R < $0;;
+        *)
+                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
+ent"
+                exit 1;;
+esac
+
+exit 0
+--- centericq-4.21.0.orig/libicq2000-0.1/src/DirectClient.cpp
++++ centericq-4.21.0/libicq2000-0.1/src/DirectClient.cpp
+@@ -162,6 +162,7 @@
+ 
+       m_recv.setLittleEndian();
+       m_recv >> length;
++      if (length == 0) return; // short read, toss it back (nothing to do)
+       if (length > Incoming_Packet_Limit) throw ParseException("Received too long incoming packet");
+       if (m_recv.remains() < length) return; // waiting for more of the packet
+ 
