--- centericq-4.21.0.orig/debian/patches/seconds_in_timestamps_cleaner.dpatch
+++ centericq-4.21.0/debian/patches/seconds_in_timestamps_cleaner.dpatch
@@ -0,0 +1,195 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## seconds_in_timestamps by Sebastien Delafond <seb@debian.org>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Add user setting to display timestamps up to the second.
+
+@DPATCH@
+diff -urNad centericq-4.21.0~/src/icqconf.cc centericq-4.21.0/src/icqconf.cc
+--- centericq-4.21.0~/src/icqconf.cc	2006-10-04 11:30:06.000000000 -0700
++++ centericq-4.21.0/src/icqconf.cc	2006-10-04 11:30:09.000000000 -0700
+@@ -41,6 +41,11 @@
+ #include "connwrap.h"
+ 
+ icqconf::icqconf() {
++    DEFAULT_TIMESTAMP_FORMAT = "DD.MM.YY hh:mm";
++    DEFAULT_LOGTIMESTAMP_FORMAT = "%R ";
++    SECONDS_TIMESTAMP_FORMAT = "DD.MM.YY hh:mm:ss";
++    SECONDS_LOGTIMESTAMP_FORMAT = "%T ";
++
+     rs = rscard;
+     rc = rcdark;
+     cm = cmproto;
+@@ -50,8 +55,11 @@
+ 
+     hideoffline = antispam = makelog = askaway = logtimestamps =
+ 	logonline = emacs = proxyssl = proxyconnect = notitles =
+-	debug = false;
++	debug = timestampstothesecond = false;
+ 
++    timestampformat = DEFAULT_TIMESTAMP_FORMAT;
++    logtimestampformat = DEFAULT_LOGTIMESTAMP_FORMAT;
++ 
+     startoffline = false;
+ 
+     savepwd = mailcheck = fenoughdiskspace = true;
+@@ -466,6 +474,7 @@
+ 	    if(param == "logonline") logonline = true; else
+ 	    if(param == "fromcharset") fromcharset = buf; else
+ 	    if(param == "tocharset") tocharset = buf; else
++	    if(param == "timestampstothesecond") settimestampstothesecond(true); else
+ 	    if(param == "ptp") {
+ 		ptpmin = atoi(getword(buf, "-").c_str());
+ 		ptpmax = atoi(buf.c_str());
+@@ -551,6 +560,7 @@
+ 
+ 	    if(!getbidi()) f << "nobidi" << endl;
+ 	    if(logtimestamps) f << "logtimestamps" << endl;
++	    if(timestampstothesecond) f << "timestampstothesecond" << endl;
+ 	    if(logonline) f << "logonline" << endl;
+ 
+ 	    f << "smtp\t" << getsmtphost() << ":" << getsmtpport() << endl;
+@@ -937,6 +947,17 @@
+     hideoffline = fho;
+ }
+ 
++void icqconf::settimestampstothesecond(bool ttts) {
++    timestampstothesecond = ttts;
++    if (timestampstothesecond) {
++        timestampformat = SECONDS_TIMESTAMP_FORMAT;
++	logtimestampformat = SECONDS_LOGTIMESTAMP_FORMAT;
++    } else {
++        timestampformat = DEFAULT_TIMESTAMP_FORMAT;
++	logtimestampformat = DEFAULT_LOGTIMESTAMP_FORMAT;
++    }
++}
++
+ void icqconf::setemacs(bool fem) {
+     emacs = fem;
+ }
+diff -urNad centericq-4.21.0~/src/icqconf.h centericq-4.21.0/src/icqconf.h
+--- centericq-4.21.0~/src/icqconf.h	2006-10-04 11:30:06.000000000 -0700
++++ centericq-4.21.0/src/icqconf.h	2006-10-04 11:30:57.000000000 -0700
+@@ -137,7 +137,8 @@
+ 
+ 	bool hideoffline, quote, savepwd, antispam, mailcheck,
+ 	    makelog, fenoughdiskspace, askaway, bidi, logtimestamps,
+-	    logonline, emacs, proxyconnect, proxyssl, notitles, debug;
++	    logonline, emacs, proxyconnect, proxyssl, notitles, debug,
++	    timestampstothesecond;
+ 
+ 	bool startoffline;
+ 
+@@ -150,6 +151,11 @@
+ 	    bindhost, httpproxyhost, httpproxyuser, httpproxypasswd,
+ 	    fromcharset, tocharset, browser;
+ 
++	char *DEFAULT_TIMESTAMP_FORMAT, *DEFAULT_LOGTIMESTAMP_FORMAT,
++	  *SECONDS_TIMESTAMP_FORMAT, *SECONDS_LOGTIMESTAMP_FORMAT;
++
++	char *timestampformat, *logtimestampformat;
++	
+ 	map<string, string> actions;
+ 
+ 	regsound rs;
+@@ -248,6 +254,12 @@
+ 	bool getnonimonline(protocolname pname);
+ 	void setnonimonline(protocolname pname, bool fnonimonline);
+ 
++	bool gettimestampstothesecond() const { return timestampstothesecond; }
++	void settimestampstothesecond(bool ftimestampstothesecond);
++
++	char *gettimestampformat() const { return timestampformat; } ;
++	char *getlogtimestampformat() const { return logtimestampformat; } ;
++
+ 	groupmode getgroupmode() const { return fgroupmode; }
+ 	void setgroupmode(groupmode amode);
+ 
+diff -urNad centericq-4.21.0~/src/icqdialogs.cc centericq-4.21.0/src/icqdialogs.cc
+--- centericq-4.21.0~/src/icqdialogs.cc	2006-10-04 11:30:06.000000000 -0700
++++ centericq-4.21.0/src/icqdialogs.cc	2006-10-04 11:30:10.000000000 -0700
+@@ -1103,7 +1103,7 @@
+     bool bidi = conf.getbidi();
+     bool emacs = conf.getemacs();
+     bool proxyconnect = conf.getproxyconnect();
+-
++    bool timestampstothesecond = conf.gettimestampstothesecond();
+     bool logtimestamps, logonline;
+     conf.getlogoptions(logtimestamps, logonline);
+ 
+@@ -1249,6 +1249,7 @@
+ 
+ 	i = t.addnode(_(" Logging "));
+ 	t.addleaff(i, 0, 9, _(" Timestamps in the log window : %s "), stryesno(logtimestamps));
++	t.addleaff(i, 0, 48, _(" Timestamps include seconds : %s "), stryesno(timestampstothesecond));
+ 	t.addleaff(i, 0, 10, _(" Online/offile events in the log window : %s "), stryesno(logonline));
+ 	t.addleaff(i, 0, 18, _(" Detailed IM events log in ~/.centericq/log : %s "), stryesno(makelog));
+ 
+@@ -1383,8 +1384,10 @@
+ 		        tmp = inputstr(_("HTTP browser to use: "), browser);
+ 			if(!tmp.empty()) browser = browser;
+ 		        break;
+-		}
+-		break;
++ 		    case 48:
++		        timestampstothesecond = !timestampstothesecond; break;
++  		}
++  		break;
+ 	    case 1:
+ 		finished = success = true;
+ 		conf.setquote(quote);
+@@ -1397,6 +1400,7 @@
+ 		conf.setmakelog(makelog);
+ 		conf.setaskaway(askaway);
+ 		conf.setproxyconnect(proxyconnect);
++		conf.settimestampstothesecond(timestampstothesecond);
+ 		conf.setcharsets(convertfrom, convertto);
+ 
+ 		for(pname = icq; pname != protocolname_size; pname++) {
+diff -urNad centericq-4.21.0~/src/icqface.cc centericq-4.21.0/src/icqface.cc
+--- centericq-4.21.0~/src/icqface.cc	2006-10-04 11:30:07.000000000 -0700
++++ centericq-4.21.0/src/icqface.cc	2006-10-04 11:30:10.000000000 -0700
+@@ -1982,7 +1982,7 @@
+ 	time_t t;
+ 	char stime[64];
+ 	time(&t);
+-	strftime(stime, 64, "%R ", localtime(&t));
++	strftime(stime, 64, conf.getlogtimestampformat(), localtime(&t));
+ 	text.insert(2, stime);
+     }
+ 
+@@ -2504,8 +2504,8 @@
+     time_t t, ts;
+     char buf[64];
+ 
+-    ds1 = time2str(&(t = ev.gettimestamp()), "DD.MM.YY hh:mm", buf);
+-    ds2 = time2str(&(ts = ev.getsenttimestamp()), "DD.MM.YY hh:mm", buf);
++    ds1 = time2str(&(t = ev.gettimestamp()), conf.gettimestampformat(), buf);
++    ds2 = time2str(&(ts = ev.getsenttimestamp()), conf.gettimestampformat(), buf);
+ 
+     r = ds1 + " ";
+     tdiff = ts-t;
+@@ -2942,9 +2942,9 @@
+ 
+ 	t = ev.gettimestamp();
+ 	ts = ev.getsenttimestamp();
+-	text = (string) + " " + time2str(&t, "DD.MM hh:mm", buf) + " ";
++	text = (string) + " " + time2str(&t, conf.gettimestampformat(), buf) + " ";
+ 	if ((t - ts) > 0) 
+-	    text += (string) + "[" + time2str(&ts, "DD.MM hh:mm", buf) + "] ";
++	    text += (string) + "[" + time2str(&ts, conf.gettimestampformat(), buf) + "] ";
+ 	text += ev.gettext();
+ 
+ 	if(ev.getdirection() == imevent::incoming) {
+diff -urNad centericq-4.21.0~/src/icqhist.cc centericq-4.21.0/src/icqhist.cc
+--- centericq-4.21.0~/src/icqhist.cc	2006-10-04 11:30:00.000000000 -0700
++++ centericq-4.21.0/src/icqhist.cc	2006-10-04 11:30:10.000000000 -0700
+@@ -294,7 +294,7 @@
+ 
+ 	    if(text.size()) {
+ 		if(text.size() > COLS) text.erase(COLS);
+-		text = (string) " " + time2str(&lastread, "DD.MM hh:mm", buf) + " " + text;
++		text = (string) " " + time2str(&lastread, conf.gettimestampformat(), buf) + " " + text;
+ 		color = dir == HIST_MSG_IN ? conf.getcolor(cp_main_text) : conf.getcolor(cp_main_highlight);
+ 
+ 		histentry h;
