$MirOS$
--- Makefile.PL.orig	Wed Dec 10 17:15:49 2008
+++ Makefile.PL	Fri Dec 26 14:11:21 2008
@@ -4,11 +4,11 @@ use 5.000;
 use ExtUtils::MakeMaker;
 
 # configuration, set to auto or manual
-my $cfg = 'auto';
+my $cfg = 'manual';
 # override these values if manual mode
-my $libs = '';
-my $defs = '-DHAVE_ARC4RANDOM_PUSHB=1';
-my $incs = '';
+my $libs = '-lbsd';
+my $defs = '-DHAVE_ARC4RANDOM_PUSHB=0';
+my $incs = '-include /usr/include/bsd/bsd.h';
 my $objs = '';
 # do not override below for auto mode
 if ($cfg eq 'auto') {
